spec/requests/invites_controller_spec.rb

```diff
@@ -245,6 +245,28 @@
     end
   end
 
+  context '#retrieve' do
+    it 'requires to be logged in' do
+      post '/invites/retrieve.json', params: { email: 'first_name@example.com' }
```

Comment:
I think this test is only failing because there is no record. We need to add the action to `requires_login`

Reply:
Compared to all the other tests I'm looking at, this looks to be correct, and tests didn't fail here, do you have more details?

Reply:
If you look here:

https://github.com/discourse/discourse/blob/master/app/controllers/invites_controller.rb#L7

You will see that other actions in the controller are added to `requires_login` so they can be authenticated. Your test is not failing, I believe, because the record you are looking for does not exist. You could confirm this by having your test run after that record is created (but still not logged in.) It should fail.

Reply:
Ah, I see, I was looking in the tests source code, sorry! I'm new to everything Ruby, pushed a fix